spec/lib/iiif_manifest/manifest_factory_spec.rb

```diff
@@ -163,6 +163,96 @@ def sequence_rendering
       end
     end
 
+    context 'where there is a no manifest_metadata method' do
+      let(:file_presenter) { DisplayImagePresenter.new }
+
+      it 'does not have a metadata element' do
+        allow(IIIFManifest::ManifestBuilder::CanvasBuilder).to receive(:new).and_call_original
+        allow(book_presenter).to receive(:file_set_presenters).and_return([file_presenter])
+        expect(result['metadata']).to eq nil
+      end
+    end
+
+    context 'where there is a manifest_metadata method' do
+      let(:file_presenter) { DisplayImagePresenter.new }
+
+      before do
+        class Book
```

Comment:
You may want to consider an after hook that destroys this class: ` Object.send(:remove_const, :Book)`

Reply:
You can avoid declaring the class entirely and just stub the manifest_metadata method.
```
    context 'where there is a manifest_metadata method' do
      let(:metadata) { [{ 'label' => 'Title', 'value' => 'Title of the Item' }] }

      it 'has metadata' do
        allow(book_presenter).to receive(:manifest_metadata).and_return(metadata)
        expect(result['metadata'][0]['label']).to eq 'Title'
        expect(result['metadata'][0]['value']).to eq 'Title of the Item'
      end
    end
```
I found that the stubbing of the file_presenter is also unnecessary in this context.